HP-160: 상품 상세페이지 기능 및 UI 구현

minseung-gang opened the pull request:
## PR 타입

- [x] 기능 추가
- [ ] 버그 수정
- [ ] 리팩토링
- [ ] 문서 수정
- [ ] 기타

## 반영 브랜치

HP-160/ms → dev

## 작업 사항
### 상품 상세 페이지 구현
ProductPage 경로 추가 및 routePath.common에 등록
- 상품 상세 페이지 전체 UI 및 기능 구현
- 상품 정보 영역 → ProductInfoSection 컴포넌트 분리
- 관련 상품 캐러셀 → RelatedProductsCarousel 컴포넌트 분리
- 구매 옵션 요약 영역 → StickySummary 컴포넌트 분리
- 모바일 전용 구매하기 버튼 → MobilePurchasePanel 컴포넌트 구현

상세 탭 UI 구현 및 관련 컴포넌트 연동

### 공통 리소스 정리
상품 상세 관련 상수(PRODUCT_DETAIL, SUBSCRIPTION_MONTH_OPTIONS) 분리
공통 로딩 스피너 → LoadingSpinner 컴포넌트로 분리 및 적용
useGetProductDetail에 productId 존재 여부 기반 enabled 옵션 추가

### 기타
route 오타 수정
불필요한 코드 제거 및 정리


### 미팅 건의 사항
- constants/locale 내부 코드가 페이지가 늘어나면서 양이 많아 질 것으로 보여서 파일별로 분리하는게 좋아보입니다!

## 체크리스트
- [x] 빌드에 성공했나요?
- [x] 코드 컨벤션을 잘 지켰나요? (예: `cmd + opt + L`)

## 테스트 결과
- 테스트 결과 이상 없습니다.

minseung-gang committed:
feat: router 경로에 ProductPage 추가

minseung-gang committed:
feat: ProductPage 컴포넌트에 필요한 상수 분리 및 정리

minseung-gang committed:
feat: ProductPage 경로를 index에서 export 하도록 추가

minseung-gang committed:
fix: product path경로 오타 수정

minseung-gang committed:
feat: routePath.common에 product 경로 추가

minseung-gang committed:
feat: 상품 상세페이지 관련 PRODUCT_DETAIL 상수 분리 및 관리

minseung-gang committed:
feat: 구독 개월 수 관련 상수 SUBSCRIPTION_MONTH_OPTIONS 추가

minseung-gang committed:
feat: useGetProductDetail에 productId 유무 기반 enabled 옵션 추가

minseung-gang committed:
refactor: LoadingSpinner를 공통 컴포넌트로 분리 및 ProductList에 적용

minseung-gang committed:
refactor: 불필요한 variant="default" 속성 제거

minseung-gang committed:
feat: 상품 상세 페이지 기능 및 UI 구현

minseung-gang committed:
feat: 상품 상세 영역을 ProductInfoSection 컴포넌트로 분리 및 적용

minseung-gang committed:
feat: 관련 상품 캐러셀 영역을 RelatedProductsCarousel 컴포넌트로 분리 및 적용

minseung-gang committed:
feat: 상품 구매 옵션을 StickySummary 컴포넌트로 분리 및 적용

minseung-gang committed:
feat: 모바일에서만 노출되는 구매하기 버튼 UI 컴포넌트 구현

minseung-gang committed:
feat: 상품 상세 탭 UI 구현 및 StickySummary, MobilePurchasePanel 연동

jieun419 commented on src/pages/product/components/organisms/MobilePurchasePanel.tsx:
```diff
@@ -0,0 +1,119 @@
+import { SUBSCRIPTION_MONTH_OPTIONS } from '@/constants/subscription';
+import useLocale from '@/hooks/useLocale';
+
+interface MobilePurchasePanel {
```
props 관련 타입 정의로 
MobilePurchasePanel -> `MobilePurchasePanelProps` 명칭이 맞을 것 같습니다!

jieun419 commented on src/pages/product/components/organisms/ProductDetailTabs.tsx:
```diff
@@ -0,0 +1,128 @@
+    { label: PRODUCT_DETAIL[currentLocale].description, value: product.description },
+  ] as const;
+
+  console.log('product', product);
```
console.log 삽입 되어있습니다! 해당 코드 삭제 필요할 것 같아요

jieun419 commented on src/constants/path.ts:
```diff
@@ -6,6 +6,7 @@ const common = {
   login: '/login',
   logout: '/logout',
   oauthRedirect: '/oauth-redirect',
+  product: '/product/:productId',
```
product의 경우 라우터 등록용, 동적으로 이동하는 path 두개로 관리하면 좋을 것 같습니다!

이후 route관련 path가 필요할 경우 하단 또는 다른 방식의 컨벤션을 따라 가면 좋을 것 같아요!

```ts
// 예시 (더 나은 구조 있다면 그 방식으로 수정해도 좋다 생각합니다.
product: {
  root: (productId: string) => 'product/${productId}',
  route: '/product/:productId',
}
```

minseung-gang replied:
오~ 좋은 방법인 것 같습니다!

jieun419 commented on src/pages/product/index.tsx:
```diff
@@ -0,0 +1,87 @@
+  const { data: bestProductData, isLoading: isLoadingRelatedProducts } = useGetRelatedProducts();
+
+  const handleProductClick = (productId: string) => {
+    navigate(`/product/${productId}`);
```
routePath에 선언 후 관리하면 좋을 것 같아요!

```ts
// ex
navigate(routePath.common.product.root(productId))
```

jieun419 commented on src/pages/product/index.tsx:
```diff
@@ -0,0 +1,87 @@
+  };
+
+  const handlePurchase = (productId: string) => {
+    navigate(`/purchase/${productId}`);
```
이전 코멘트 처럼 routePath에 선언 후 관리하면 좋을 것 같아요!

jieun419 commented on src/pages/product/components/molecules/ProductInfoSection.tsx:
ProductInfoSection에서 썸네일, 개월 옵션 등의 컴포넌트는 atom으로 나눠 관리하면 구조가 명확해지고 가독성이나 유지보수 측면에서도 유리할 것 같습니다!
혹시 이부분에 대해 어떻게 생각하신가요?

minseung-gang replied:
[refactor: ProductInfoSection 컴포넌트 구조 개선](https://github.com/happy-pill/happyPill_Front/pull/12/commits/54b67243bc544a5a48e2d2abedcd84d03e1f6a66)
이렇게 수정했습니다! 아토믹 디자인 패턴에 맞게 폴더생성 및 적용했습니다.

jieun419 commented on src/pages/product/index.tsx:
```diff
@@ -0,0 +1,87 @@
+  const { productId } = useParams();
+  const [subscriptionOption, setSubscriptionOption] = useState(1);
+  console.log('script', subscriptionOption);
+  /*  const [relatedProducts, setRelatedProducts] = useState<RelatedProduct[]>([]) */
```
16,17번줄 console.log와 주석된 코드가 포함되어 있습니다! 해당 코드 삭제 필요합니다!

주석된 코드는 추후 사용되는 코드라면 코멘트를 추가해 주시면 좋을 것 같아요!

jieun419 reviewed (commented):
고생하셨습니다! 👍

- 콘솔, 주석 코드가 몇 군데 존재합니다! (불필요한 코드라면 삭제해도 좋을 것 같아요!)
- routePath에서 route 등록, 동적 path 두 가지를 같이 관리하면 좋을 것 같습니다! 
(애매 하다면 미팅 때 같이 논의해도 좋을 것 같습니다!)


----


>constants/locale 내부 코드가 페이지가 늘어나면서 양이 많아 질 것으로 보여서 파일별로 분리하는게 좋아보입니다!

저도 동의 합니다! `constants/locale` 에서 기능별 또는 페이지 단위로 파일을 한번 더 나누면 확장성 측면에서도 좋을 것 같아요!

minseung-gang committed:
refactor: MobilePurchasePanel의 props 타입명을 MobilePurchasePanelProps로 변경

minseung-gang committed:
chore: 불필요한 console.log 제거

minseung-gang committed:
refactor: product route를 동적 경로와 라우터 등록용으로 분리

minseung-gang committed:
feat: 결제 페이지 경로를 직접구매/장바구니용으로 분리하여 routePath에 추가

minseung-gang committed:
refactor: 불필요한 코드 삭제 및 routePath 적용

minseung-gang committed:
feat: ProductInfoSection에서 컴포넌트 분할 및 적용

- ProductThumbnail 컴포넌트 분리
- SubscriptionOptions 컴포넌트 분리
- ProductHeader 컴포넌트 분리
- ProductDescription 컴포넌트 분리

minseung-gang committed:
refactor: ProductInfoSection 컴포넌트 구조 개선

minseung-gang commented:
[feat: 결제 페이지 경로를 직접구매/장바구니용으로 분리하여 routePath에 추가](https://github.com/happy-pill/happyPill_Front/pull/12/commits/065df33404d6f38cb313871d2562a3deb194904f)
[refactor: product route를 동적 경로와 라우터 등록용으로 분리](https://github.com/happy-pill/happyPill_Front/pull/12/commits/df4304c64c0dc09e8e73fb92ac0d415c9cfc5f58)

paths 파일 내부에 해당 상품, 장바구니 페이지의 routePath를 구성했습니다.
좋은 코드리뷰 감사드립니다 지은님!

minseung-gang committed:
refactor: routePath.common.product → product.root로 router 경로 명확화

jieun419 commented on src/pages/product/components/molecules/ProductInfoSection.tsx:
```diff
@@ -49,55 +52,30 @@ const ProductInfoSection: React.FC<productInfoProps> = ({
-        className='h-[150px] w-full rounded-md border-[#E2E2E2] object-cover md:h-auto'
-        alt={`${product?.name} `}
-      />
+      <ProductThumbnail src={product?.thumbnailUrl} alt={`${product?.name}`} />
```
`${}`으로 데이터 감쌀 필요 없이 바로 넣어줘도 될 것 같아요!
```
// 전
alt={`${product?.name}`}

// 후
alt={product?.name}
```

jieun419 reviewed (dismissed):
route -> 페이지를 동적으로 넘길 경우로 이해 하고
저도 해당 규칙 따라가겠습니다!

---

한군데만 수정 후 바로 머지하시면 될 것 같습니다!
고생하셨습니다!

minseung-gang committed:
refactor: alt 속성에서 불필요한 템플릿 리터럴 제거
